src/main/java/org/elasticsearch/common/jna/Natives.java

```diff
@@ -62,6 +65,42 @@ public static void tryMlockall() {
         }
     }
 
+    public static void tryVirtualLock()
+    {
+        Kernel32Library kernel = Kernel32Library.getInstance();
+        Pointer process = null;
+        try {
+            process = kernel.GetCurrentProcess();
+            // By default, Windows limits the number of pages that can be locked.
+            // Thus, we need to first increase the working set size of the JVM by
+            // the amount of memory we wish to lock (Xmx), plus a small overhead (1MB).
+            SizeT size = new SizeT(JvmInfo.jvmInfo().getMem().getHeapInit().getBytes() + (1024 * 1024));
```

Comment:
@gmarz @kimchy but it is okay for them to be different simply because we allow it :) 

The worse case is that we kill starting with MinHeap (growing to max immediately) instead of otherwise the worse case being mlockall'ing too little.